web/src/ApiButton.tsx

```diff
@@ -521,29 +521,41 @@ function ApiSubmitButton(props: PropsWithChildren<ApiButtonElementProps>) {
 //    options used on submit.
 export function ApiButton(props: PropsWithChildren<ApiButtonProps>) {
   const { className, uiButton, ...buttonProps } = props
+  const buttonName = uiButton.metadata?.name || ""
 
   const [inputValues, setInputValues] = usePersistentState<{
     [name: string]: any
-  }>(`apibutton-${uiButton.metadata?.name}`, {})
+  }>(`apibutton-${buttonName}`, {})
   const { enqueueSnackbar } = useSnackbar()
   const pb = usePathBuilder()
   const { setError } = useHudErrorContext()
 
   const [loading, setLoading] = useState(false)
-  const [confirming, setConfirming] = useState(false)
+  // Record the confirmation state with the current button's name,
+  // so if the button name changes, the confirmation state can be reset
+  const [confirmingByButton, setConfirmingByButton] = useState({
```

Comment:
i...don't think this is how state is intended to be used (where you use the initial value to communicate props changes)

have you tried something like: 

```
useEffect(() => setConfirming(false), [buttonName])
```

That will reset the confirmation state when buttonName changes, which i think is closer to what you want.

Reply:
(one of my big frustrations with react is that it has tons of undefined/underdetermined behavior if you don't use state in exactly the way it expects)

Reply:
i think `useLayoutEffect` might work instead. i stayed away from `useEffect` because it runs after the render is painted and i thought any state changes were discouraged from it (which i can't find again in the react docs, so perhaps i made that up!) i'll update in a moment.